crates/re_viewer/src/ui/selection_panel.rs

```diff
@@ -413,6 +408,226 @@ fn entity_props_ui(
         });
 }
 
+fn visible_history_ui(
+    ctx: &mut ViewerContext<'_>,
+    ui: &mut Ui,
+    space_view_class_name: &SpaceViewClassName,
+    entity_path: Option<&EntityPath>,
+    entity_props: &mut EntityProperties,
+) {
+    //TODO(#4107): support more space view types.
+    if space_view_class_name != "3D" && space_view_class_name != "2D" {
+        return;
+    }
+
+    if !should_display_visible_history(ctx, entity_path) {
+        return;
+    }
+
+    let re_ui = ctx.re_ui;
+
+    re_ui.checkbox(
+        ui,
+        &mut entity_props.visible_history.enabled,
+        "Visible history",
+    ).on_hover_text(
+        "Enable the Visible History feature.\n\nBy default, only the last data received before the \
+        current time is shown. By activating the Visible History, all the data received within a \
+        time window is shown instead.");
```

Comment:
```suggestion
        "Enable Visible History.\n\nBy default, only the last state before the \
        current time is shown. By activating Visible History, all data within a \
        time window is shown instead.");
```
maybe just personal preference, but I find it easier to read this way. Don't want to talk about 'received' because that implies network which is optional.